### PR TITLE
Update to Chromium 112.0.5615.49/50

### PR DESCRIPTION
**Pull request by Ahrotahn**
This PR updates to Chromium 112.0.5615.49/50.

Notes:
* A set of exclusions were removed from update_lists.py due to the removal of TabRanker in [[4281937]](https://chromium-review.googlesource.com/c/chromium/src/+/4281937)

* `enable_js_type_check` has been removed from the build flags: [[4210955]](https://chromium-review.googlesource.com/c/chromium/src/+/4210955)

* A new addition to disable-google-host-detection.patch for [[4225611]](https://chromium-review.googlesource.com/c/chromium/src/+/4225611)

* The removal for chrome_content_browser_client_receiver_bindings.cc in the safebrowsing patch is due to that section being gated with safebrowsing buildflags in [[4024889]](https://chromium-review.googlesource.com/c/chromium/src/+/4024889)

* Some new additions to the safebrowsing patch to handle new changes to the file analyzers in [[4235459]](https://chromium-review.googlesource.com/c/chromium/src/+/4235459).  This makes the Debian unrar patch unnecessary and will likely need to be removed once the changes in [[4350049]](https://chromium-review.googlesource.com/c/chromium/src/+/4350049) land.

* A new addition to remove-unused-preferences-fields.patch for the changes in [[4193805]](https://chromium-review.googlesource.com/c/chromium/src/+/4193805)

* A new cloud management error is being logged due to an addition in [[4221504]](https://chromium-review.googlesource.com/c/chromium/src/+/4221504)

<br>

[Change log](https://chromium.googlesource.com/chromium/src/+log/111.0.5563.148..112.0.5615.50?pretty=fuller&n=10000)
[Blog post](https://chromereleases.googleblog.com/2023/04/stable-channel-update-for-desktop.html)

### CODE REVIEW

**Comment by macchrome** [comment]
disable-google-host-detection.patch might need adjusting/amending:

**Windows x64 cross-build**

[20/34] LINK(DLL) chrome.dll chrome.dll.lib chrome.dll.pdb
FAILED: chrome.dll chrome.dll.lib chrome.dll.pdb
../../third_party/llvm-build/Release+Asserts/bin/lld-link --rsp-quoting=posix /OUT:./chrome.dll /nologo -libpath:../../third_party/llvm-build/Release+Asserts/lib/clang/17/lib/windows /winsysroot:../../third_party/depot_tools/win_toolchain/vs_files/8649a5ff1f /MACHINE:X64 /IMPLIB:./chrome.dll.lib /DLL /PDB:./chrome.dll.pdb @./chrome.dll.rsp
lld-link: error: undefined symbol: char const *const variations::kClientDataHeader
>>> referenced by ../../components/variations/net/variations_http_headers.cc
>>>               obj/components/variations/net/net/variations_http_headers.obj
[21/34] ACTION //chrome/test/chromedriver:generate_chromedriver_license(//build/toolchain/win:win_clang_x64)
ninja: build stopped: subcommand failed.


disable-google-host-detection.patch

components/variations/net/variations_http_headers.cc

bool GetVariationsHeader(const network::ResourceRequest& request,
                         std::string* out) {
  return false;
//  return request.cors_exempt_headers.GetHeader(kClientDataHeader, out);    <<<<
}

bool ShouldAppendVariationsHeaderForTesting(
    const GURL& url,
    const std::string& histogram_suffix) {
     return false;
//      return ShouldAppendVariationsHeader(url, histogram_suffix);          <<<< Perhaps just remove function!!!
}

  ./components/variations/net/variations_http_headers.cc(116,6): warning: unused function 'ShouldAppendVariationsHeader' [-Wunused-function

**Comment by teeminus** [comment]
Builds and runs fine on windows with the fixed patch.

**Comment by PF4Public** [comment]
Windows wants .50

**Comment by Ahrotahn** [comment]
The tarball for .50 won't be generated until a couple more hours from now.  I'll tack it on once it's available.

File: devutils/update_lists.py [hunk 1]
```diff
@@ -367,7 +363,8 @@ def main(args_list=None):
         dest='error_unused',
         help='Do not treat unused patterns/prefixes as an error.')
     args = parser.parse_args(args_list)
-    DOMAIN_EXCLUDE_PREFIXES.extend(args.domain_exclude_prefix)
+    if args.domain_exclude_prefix is not None:
+        DOMAIN_EXCLUDE_PREFIXES.extend(args.domain_exclude_prefix)
```
Comment (PF4Public) [inline devutils/update_lists.py:367]: This make this line work only if exclude prefix not set, right?

**Review by networkException — APPROVED** [review]

**Pull request merged**